test/unit_test/CMakeLists.txt

```diff
@@ -12,20 +12,77 @@ add_library(
 target_compile_definitions(UNITTEST_Util PRIVATE ${QUARTZ_COMPILE_DEFINITIONS})
 target_compile_options(UNITTEST_Util PRIVATE ${QUARTZ_CMAKE_CXX_FLAGS})
 target_include_directories(UNITTEST_Util PRIVATE ${QUARTZ_UNIT_TEST_INCLUDE_DIRS} )
-target_link_libraries(UNITTEST_Util PUBLIC UTIL_Logger)
+target_link_libraries(
+    UNITTEST_Util
+
+    PUBLIC
+    UTIL_Logger
+
+    PUBLIC
+    QUARTZ_MANAGERS_PhysicsManager
+)
+
+#====================================================================
+# Functionality for easily creating unit tests
+#====================================================================
+
+function(create_unit_test unit_test_name unit_test_main_file unit_test_link_libraries)
+    add_executable(${unit_test_name} ${unit_test_main_file})
+    target_compile_definitions(${unit_test_name} PRIVATE ${QUARTZ_COMPILE_DEFINITIONS})
+    target_compile_options(${unit_test_name} PRIVATE ${QUARTZ_CMAKE_CXX_FLAGS})
+    target_include_directories(${unit_test_name} PRIVATE ${QUARTZ_UNIT_TEST_INCLUDE_DIRS} )
+    target_link_libraries(${unit_test_name} PRIVATE ${unit_test_link_libraries} UNITTEST_Util)
+    add_test(NAME ${unit_test_name} COMMAND ${unit_test_name})
+endfunction()
```

Comment:
derive `unit_test_main_file` from the `unit_test_name` + the provided directory, this will allow us to only write the full name once

additionally we should allow the final parameter to be a list. This may result in the whole argument declaration in being a VA_ARGS type of situation, if that is the case we can do the following:
- ensure there are at least three arguments
- the first argument is the name of the unit test main file
- the second argument is the directory in which the unit test main file lives
- the third, and subsequent arguments get turned into a list representing the link libraries